src/unstract/sdk/vector_db.py

```diff
@@ -62,19 +83,87 @@ def get_vector_db(
                 Common.METADATA
             ][Common.ADAPTER]
             vector_db_metadata = vector_db_config.get(Common.ADAPTER_METADATA)
-            org = self.__get_org_id()
+            org = self._get_org_id()
             # Adding the collection prefix and embedding type
             # to the metadata
             vector_db_metadata[VectorDbConstants.VECTOR_DB_NAME] = org
             vector_db_metadata[
                 VectorDbConstants.EMBEDDING_DIMENSION
-            ] = embedding_dimension
+            ] = self._embedding_dimension
 
-            vector_db_adapter_class = vector_db_adapter(vector_db_metadata)
-            return vector_db_adapter_class.get_vector_db_instance()
+            self.vector_db_adapter_class = vector_db_adapter(vector_db_metadata)
+            return self.vector_db_adapter_class.get_vector_db_instance()
         except Exception as e:
-            self.tool.stream_log(
-                log=f"Unable to get vector_db {adapter_instance_id}: {e}",
+            self._tool.stream_log(
+                log=f"Unable to get vector_db {self._adapter_instance_id}: {e}",
                 level=LogLevel.ERROR,
             )
-            raise ToolVectorDBError(f"Error getting vectorDB instance: {e}") from e
+            raise VectorDBError(f"Error getting vectorDB instance: {e}") from e
+
+    def get_vector_store_index_from_storage_context(
+        self,
+        documents: Sequence[Document],
+        storage_context: Optional[StorageContext] = None,
+        show_progress: bool = False,
+        **kwargs,
+    ) -> IndexType:
+        if not self._embedding_instance:
+            raise VectorDBError("Vector DB does not have an embedding initialised")
+        parser = kwargs.get("node_parser")
+        return VectorStoreIndex.from_documents(
+            documents,
+            storage_context=storage_context,
+            show_progress=show_progress,
+            embed_model=self._embedding_instance,
+            node_parser=parser,
+        )
+
+    def get_vector_store_index(self, **kwargs: Any) -> VectorStoreIndex:
+        if not self._embedding_instance:
+            raise VectorDBError("Vector DB does not have an embedding initialised")
+        return VectorStoreIndex.from_vector_store(
+            vector_store=self._vector_db_instance,
+            embed_model=self._embedding_instance,
+            kwargs=kwargs,
+        )
+
+    def get_storage_context(self) -> StorageContext:
+        return StorageContext.from_defaults(vector_store=self._vector_db_instance)
+
+    def query(self, query) -> VectorStoreQueryResult:
```

Comment:
@gaya3-zipstack Better to have a thin wrapper on top and not expose LI constructs directly.